up42/http/oauth.py

```diff
@@ -93,3 +94,28 @@ def token(self) -> Token:
         if self._token.has_expired:
             self._token = self._fetch_token()
         return self._token
+
+
+def detect_settings(credentials: dict) -> Optional[config.CredentialsSettings]:
+    if all(credentials.values()):
+        keys = credentials.keys()
+        if keys == {"project_id", "project_api_key"}:
+            warnings.warn(
+                "Project based authentication will be deprecated."
+                "Please follow authentication guidelines (/docs/authentication.md)."
+            )
+            return config.ProjectCredentialsSettings(**credentials)
+        if keys == {"username", "password"}:
+            return config.AccountCredentialsSettings(**credentials)
+        raise InvalidCredentials(f"Credentials {credentials} are not supported")
```

Comment:
just pointing out - this will expose the credentials. Even though the user themself would be aware of those credentials already, should we just expose the keys instead?

Reply:
In this case will expose the invalid credentials. But can be something to consider as the user can have a typo.